lib/logstash/inputs/google_pubsub.rb

```diff
@@ -265,16 +265,18 @@ def run(queue)
         end
 
         ack_ids = messages.map{ |msg| msg["ackId"] }
-        result = request(
-          :api_method => @pubsub.projects.subscriptions.acknowledge,
-          :parameters => {'subscription' => @subscription},
-          :body_object => {
-            :ackIds => ack_ids
-          }
-        )
-        if result.error?
-          @logger.error("Error #{result.status}: #{result.error_message}")
-        end
+        if ack_ids.any?
```

Comment:
alternately, on line 252 where we decide based on the payload whether to do anything:

~~~ ruby
       if messages
~~~

we could replace with:
~~~ ruby
       if messages && !messages.empty?
~~~

This would ensure that we only process a batch at all if it contained at least one message; from what I understand of the problem, we already effectively skip the first part of the processing step when we send `Enumerable#each { ... }` to an empty `messages` (a no-op operation), but when we don't have any messages we don't need to send the `ACK`s either.

Reply:
```if messages && !messages.empty?``` could be replaced with ```if messages.any?``` as ```JSON.parse(result.body)``` on line 244 will return an empty array. I'll update the PR